ludwig/decoders/sequence_decoders.py

```diff
@@ -13,18 +13,227 @@
 # limitations under the License.
 # ==============================================================================
 import logging
+from typing import Dict, Tuple
 
-from ludwig.constants import SEQUENCE, TEXT
+import torch
+import torch.nn as nn
+
+from ludwig.constants import HIDDEN, LOGITS, SEQUENCE, TEXT
 from ludwig.decoders.base import Decoder
 from ludwig.decoders.registry import register_decoder
+from ludwig.utils import strings_utils
 
 logger = logging.getLogger(__name__)
 
 
-# TODO(Justin): Implement sequence decoders.
+class RNNDecoder(nn.Module):
```

Comment:
Should the super class be `LudwigModule`?  Same comment will apply to the other classes added.

Reply:
LudwigModule has additional methods like `input_shape` and `output_shape`.

I'm open to the idea, but I don't see too much benefit over the standard `torch.nn.Module` for these Decoder-internal classes.

Using `torch.nn.Module` also makes our implementation more easily referenceable against other open source versions of these models.

@ShreyaR @w4nderlust WDYT?

Reply:
@justinxzhao   Recalling our conversation, I think you're saying these classes are completely encapsulated w/in the `SequenceGeneratorDecoder` class, for example, and will never be visible to the "outside world".  If this is the case, then what I wrote may not make sense.

Reply:
I second @jimthompson5802 's last comment